packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -198,14 +205,18 @@ contract SystemConfig is OwnableUpgradeable, ReinitializableBase, ISemver {
         _setResourceConfig(_config);
 
         l2ChainId = _l2ChainId;
+        superchainConfig = ISuperchainConfig(_superchainConfig);
     }
 
-    /// @notice Upgrades the SystemConfig by setting the L2 chain ID variable.
-    /// @param _l2ChainId The L2 chain ID that this SystemConfig configures.
-    function upgrade(uint256 _l2ChainId) external reinitializer(initVersion()) {
+    /// @notice Upgrades the SystemConfig by adding a reference to the SuperchainConfig.
+    /// @param _superchainConfig The SuperchainConfig contract address.
+    function upgrade(uint256 _l2ChainId, ISuperchainConfig _superchainConfig) external reinitializer(initVersion()) {
         // Set the L2 chain ID.
         l2ChainId = _l2ChainId;
```

Comment:
`l2ChainId` was removed from the natspec params, should it be removed from the upgrade function or has it not yet been included in a previous upgrade?

Reply:
Bug, l2ChainId should be added back to the natspec params.

Reply:
To my knowledge, I just need to add it into the natspec again.